packages/cli/src/commands/init/init.js

```diff
@@ -39,25 +39,45 @@ function adjustNameIfUrl(name, cwd) {
   return name;
 }
 
-async function createFromExternalTemplate(
+async function createFromTemplate({
+  projectName,
+  templateName,
+  version,
+  npm,
+}: {
   projectName: string,
   templateName: string,
+  version?: string,
   npm?: boolean,
-) {
-  logger.debug('Initializing new project from external template');
+}) {
+  logger.debug('Initializing new project');
   logger.log(banner);
-
   const Loader = getLoader();
-
   const loader = new Loader({text: 'Downloading template'});
+
   loader.start();
 
   const templateSourceDir = fs.mkdtempSync(
     path.join(os.tmpdir(), 'rncli-init-template-'),
   );
   try {
-    let {uri, name} = await processTemplateName(templateName);
+    let uri, name;
+    if (version) {
+      if (semver.valid(version) && !semver.gte(version, '0.60.0')) {
+        throw new Error(
+          'Cannot use React Native CLI to initialize project with version lower than 0.60.0',
+        );
+      }
+      const processed = await processTemplateName(`${templateName}@${version}`);
```

Comment:
Why not pass the `version` as an argument here and deal with the error on line 67 inside of this function? If you do that you can keep doing `let {uri, name} = await processTemplate...` instead of `if...else`.

Reply:
Good catch! I wouldn't put it inside `processTemplate` myself, but move the validation logic upwards.

```
if (version && semver.valid(version) && !semver.gte(version, '0.60.0')) {
throw new Error(
    'Cannot use React Native CLI to initialize project with version lower than 0.60.0',
);
}
let {uri, name} = await processTemplateName(version ? `${templateName}@${version}` : templateName);

Reply:
Also, I don't think we should check for `semver.valid`. That can potentially swallow errors when invalid version is passed. 

I would only do `version && semver.gte` here and let `semver.gte` [throw a TypeError when invalid](https://github.com/npm/node-semver/blob/5fb517b2906a0763518e1941a3f4a163956a81d3/semver.js#L305)

Reply:
This required more work, as it turned out we don't handle versioned dependencies correctly. Added support for that, so we can now call `init --template name@1.2.3` and treat `react-native` with `--version` passed in such a way